From 6e4dc06adc689819011520cb984b97eb72505be1 Mon Sep 17 00:00:00 2001
From: Klaus Kurzmann <mok@fluxnetz.de>
Date: Sat, 13 Dec 2008 10:47:07 +0100
Subject: [PATCH] sysvinit: Simplify rc and rcS.

Remove support for boot splash.  Source initscripts end with .sh.

Author: Chia-I Wu <olv@openmoko.com>
Signed-off-by: Klaus Kurzmann <mok@fluxnetz.de>
---
 packages/sysvinit/sysvinit/openmoko/rc  |   77 +-----------------------------
 packages/sysvinit/sysvinit/openmoko/rcS |    9 ++++
 packages/sysvinit/sysvinit_2.86.bb      |    2 +-
 3 files changed, 13 insertions(+), 75 deletions(-)
 create mode 100644 packages/sysvinit/sysvinit/openmoko/rcS

diff --git a/packages/sysvinit/sysvinit/openmoko/rc b/packages/sysvinit/sysvinit/openmoko/rc
index 9d28a68..57bf3a8 100644
--- a/packages/sysvinit/sysvinit/openmoko/rc
+++ b/packages/sysvinit/sysvinit/openmoko/rc
@@ -18,22 +18,6 @@
 . /etc/default/rcS
 export VERBOSE
 
-startup_progress() {
-    step=$(($step + $step_change))
-    if [ "$num_steps" != "0" ]; then
-        progress=$((($step * $progress_size / $num_steps) + $first_step))
-    else
-        progress=$progress_size
-    fi
-    #echo "PROGRESS is $progress $runlevel $first_step + ($step of $num_steps) $step_change $progress_size"
-# FIXME: this should be virtualised and come from a generic boot splash script
-# provided by some generic splash handler - be it psplash, exquisite or
-# whatever else soa  simple install of one splash instead of another works
-    if type exquisite-write >/dev/null 2>&1; then
-         EXQUISITE_IPC=/mnt/.exquisite/exquisite exquisite-write "PROGRESS $progress" || true
-    fi
-}
-
 
 #
 # Start script or program.
@@ -45,18 +29,14 @@ startup() {
   case "$1" in
 	*.sh)
 		# Source shell script for speed.
-		(
-			trap - INT QUIT TSTP
-			scriptname=$1
-			shift
-			. $scriptname
-		)
+		scriptname=$1
+		shift
+		. $scriptname
 		;;
 	*)
 		"$@"
 		;;
   esac
-  startup_progress
 }
 
   # Ignore CTRL-C only in this shell, so we can interrupt subprocesses.
@@ -83,48 +63,6 @@ startup() {
   # Is there an rc directory for this new runlevel?
   if [ -d /etc/rc$runlevel.d ]
   then
-	# Find out where in the progress bar the initramfs got to.
-	PROGRESS_STATE=0
-	#if [ -f /dev/.initramfs/progress_state ]; then
-	#    . /dev/.initramfs/progress_state
-	#fi
-
-	# Split the remaining portion of the progress bar into thirds
-	progress_size=$(((100 - $PROGRESS_STATE) / 3))
-
-	case "$runlevel" in
-		0|6)
-			# Count down from -100 to 0 and use the entire bar
-			first_step=-100
-			progress_size=100
-			step_change=1
-			;;
-	        S)
-			# Begin where the initramfs left off and use 2/3
-			# of the remaining space
-			first_step=$PROGRESS_STATE
-			progress_size=$(($progress_size * 2))
-			step_change=1
-			;;
-		*)
-			# Begin where rcS left off and use the final 1/3 of
-			# the space (by leaving progress_size unchanged)
-			first_step=$(($progress_size * 2 + $PROGRESS_STATE))
-			step_change=1
-			;;
-	esac
-
-	num_steps=0
-	for s in /etc/rc$runlevel.d/[SK]*; do
-            case "${s##/etc/rc$runlevel.d/S??}" in
-                gdm|xdm|kdm|reboot|halt)
-                    break
-                    ;;
-            esac
-            num_steps=$(($num_steps + 1))
-        done
-        step=0
-
 	# First, run the KILL scripts.
 	if [ $previous != N ]
 	then
@@ -170,12 +108,3 @@ startup() {
 	done
   fi
 
-  # FIXME: this should be virtualised and come from a generic boot splash script
-  # provided by some generic splash handler - be it psplash, exquisite or
-  # whatever else soa  simple install of one splash instead of another works
-  if [ "x$runlevel" != "xS" ]; then
-	if type exquisite-write >/dev/null 2>&1; then
-		EXQUISITE_IPC=/mnt/.exquisite/exquisite exquisite-write "QUIT" || true
-		umount /mnt/.exquisite
-	fi
-  fi
diff --git a/packages/sysvinit/sysvinit/openmoko/rcS b/packages/sysvinit/sysvinit/openmoko/rcS
new file mode 100644
index 0000000..08dca5d
--- /dev/null
+++ b/packages/sysvinit/sysvinit/openmoko/rcS
@@ -0,0 +1,9 @@
+#!/bin/sh
+#
+# rcS		Call all S??* scripts in /etc/rcS.d in
+#		numerical/alphabetical order.
+#
+# Version:	@(#)/etc/init.d/rcS  2.76  19-Apr-1999  miquels@cistron.nl
+#
+
+exec /etc/init.d/rc S
diff --git a/packages/sysvinit/sysvinit_2.86.bb b/packages/sysvinit/sysvinit_2.86.bb
index c399874..54bb017 100644
--- a/packages/sysvinit/sysvinit_2.86.bb
+++ b/packages/sysvinit/sysvinit_2.86.bb
@@ -2,7 +2,7 @@ DESCRIPTION = "System-V like init."
 SECTION = "base"
 LICENSE = "GPL"
 HOMEPAGE = "http://freshmeat.net/projects/sysvinit/"
-PR = "r44"
+PR = "r44.1"
 
 # USE_VT and SERIAL_CONSOLE are generally defined by the MACHINE .conf.
 # Set PACKAGE_ARCH appropriately.
-- 
1.6.0.4

